### PR TITLE
Convert to use cmake-js and pkg-prebuilds

### PR DESCRIPTION
**Pull request by LandryNorris**
CMake has much better IDE support (allowing for development in VSCode and CLion). WPILib is also looking at moving C builds to CMake, so using it here may make pulling CANBridge and WPILib easier in the future.

### CODE REVIEW

**Comment by LandryNorris** [comment]
Note: I still need to take a look at how to best work with #30.

File: package.json [hunk 1]
```diff
@@ -24,10 +25,11 @@
         "typescript": "^5.0.2"
     },
     "scripts": {
-        "install": "node-gyp-build \"node scripts/download-CanBridge.mjs\"",
-        "prepublishOnly": "node scripts/download-CanBridge.mjs && tsc && prebuildify --napi",
-        "pretest": "node-gyp-build && tsc",
-        "test": "node --napi-modules test/test_binding.js"
+        "install": "pkg-prebuilds-verify binding-options.cjs || npm run build",
+        "build": "node scripts/download-CanBridge.mjs && cmake-js compile && tsc",
+        "prepublishOnly": "npm run build && pkg-prebuilds-copy --baseDir build/Release --source node_canbridge.node --name node_canbridge --napi_version=8",
```
Comment (LandryNorris) [inline package.json:30]: Other platforms besides windows need baseDir set to build, if I imagine correctly

File: package.json
Comment (NoahAndrews) [inline package.json:1]: Please remove `prebuildify` and `node-gyp` from `devDependencies`

**Review by NoahAndrews — CHANGES_REQUESTED** [review]

File: package.json
Comment (LandryNorris) [reply package.json:1]: removed them

**Comment by NoahAndrews** [comment]
Please update the lockfile

**Review by NoahAndrews — APPROVED** [review]

**Pull request merged**